feat: add "React (Atom)" to embed generator and rename rest

retrogtx opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Fixes #19679 (GitHub issue number)
- Fixes CAL-5250 (Linear issue number - should be visible at the bottom of the GitHub issue description)



https://github.com/user-attachments/assets/8945913b-5a24-46f4-bde1-c18df1e86dfb

retrogtx committed:
fix: trigger lingo.dev by removing duplicate value

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
feat: add support for React Atom embed options

retrogtx committed:
fix: update locator for React code tab in embed code generator E2E tests

retrogtx committed:
fix: update embed tab names to clarify iframe usage

retrogtx committed:
fix failing test

hariombalhara commented:
When selecting React(Atom), React(iframe) also shows as selected
![image](https://github.com/user-attachments/assets/c5b2849d-e591-45a3-b4fd-f79c6679c480)

hariombalhara commented:
In the comments, it asks to install wrong package(the iframe one) for atom

![image](https://github.com/user-attachments/assets/4c0f871b-2dab-4dcc-8c4a-4457cb0a439c)

hariombalhara commented:
Indentation of comments seems to be broken for React(iframe) 
![image](https://github.com/user-attachments/assets/ff8e6402-b10b-4670-b535-82d0f8b85f76)

hariombalhara commented:
![image](https://github.com/user-attachments/assets/94f00886-899f-4992-bf3c-cce65b2bb482)

There is unintented newlines in the React(Atom) generated code

hariombalhara commented:
I think in case of Atom, we should clearly mention that the preview at the bottom is generated using iframe approach

Somewhere above Month(April) could be a good place to mention.
![image](https://github.com/user-attachments/assets/16e03db6-7976-4ba1-b5b5-41f3a4864fad)

hariombalhara reviewed (changes requested):
Left some blocking comments !!

hariombalhara commented:
Also, I believe not every option shown on the right sidebar are supported with Atoms, we need to confirm and only show supported options.

We need to also make sure that the correct code is generated for atoms(i.e. there could possibly be different props name for Atoms maybe)

retrogtx commented:
> When selecting React(Atom), React(iframe) also shows as selected ![image](https://private-user-images.githubusercontent.com/1780212/431745355-c5b2849d-e591-45a3-b4fd-f79c6679c480.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQxOTY2MTgsIm5iZiI6MTc0NDE5NjMxOCwicGF0aCI6Ii8xNzgwMjEyLzQzMTc0NTM1NS1jNWIyODQ5ZC1lNTkxLTQ1YTMtYjRmZC1mNzljNjY3OWM0ODAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MDlUMTA1ODM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzA0NDUzZThkNjlhNjUzYTQyMzkxNzNmYzNkNjA1OWEwZWE2YTIwNzU1MGY1YWM4MzcxNWM5NDFhNzdiODA4ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.EpPHJAIzPUvtVHauXj-B65Ny_92QgXZM5wJ2nFdY5F4)

strange, looks fine on my machine

will fix the rest!

retrogtx committed:
fix: indentation and installation commands for atoms

retrogtx committed:
fix: add note about iframe usage in Preview tab

retrogtx committed:
refactor: replace Cal component with BookerEmbed in EmbedCodes

retrogtx committed:
fix: Add EmbedConfig type import in Embed.tsx

- Added missing EmbedConfig type to the existing types import
- Resolves TypeScript error "Cannot find name 'EmbedConfig'"

retrogtx committed:
fix: update BookerEmbed usage in EmbedCodes component

retrogtx committed:
fix: enhance BookerEmbed implementation and update installation instructions in EmbedTabs

retrogtx committed:
fix: improve comments in EmbedCodes for clarity on layout options and styling references

retrogtx committed:
fix: streamline BookerEmbed usage in EmbedCodes with improved class handling and layout options

hariombalhara commented:
> > When selecting React(Atom), React(iframe) also shows as selected ![image](https://private-user-images.githubusercontent.com/1780212/431745355-c5b2849d-e591-45a3-b4fd-f79c6679c480.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQxOTY2MTgsIm5iZiI6MTc0NDE5NjMxOCwicGF0aCI6Ii8xNzgwMjEyLzQzMTc0NTM1NS1jNWIyODQ5ZC1lNTkxLTQ1YTMtYjRmZC1mNzljNjY3OWM0ODAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MDlUMTA1ODM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzA0NDUzZThkNjlhNjUzYTQyMzkxNzNmYzNkNjA1OWEwZWE2YTIwNzU1MGY1YWM4MzcxNWM5NDFhNzdiODA4ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.EpPHJAIzPUvtVHauXj-B65Ny_92QgXZM5wJ2nFdY5F4)
> 
> strange, looks fine on my machine
> 
> will fix the rest!
This seems to be when selecting the embed for event-type and doesn't happen on Routing Form which you tried in the demo 
https://www.loom.com/share/e87987f4b22b444b831e6398b43e1eea?from_recorder=1&focus_title=1

hariombalhara commented on packages/features/embed/Embed.tsx:
```diff
@@ -1119,30 +1119,27 @@ const EmbedTypeCodeAndPreviewDialogContent = ({
+                              // Ensure layout is updated in config for all embed types
+                              const newConfig = (currentConfig?: EmbedConfig) => ({
+                                ...(currentConfig ?? {}),
+                                layout: option.value,
```
Looks like accidenta change of updating `layout` instead of `theme` @retrogtx

hariombalhara commented:
https://www.loom.com/share/d7ef3a4d70b0410ca219a1b47b6aff18
layout change isn't reflecting in Atom.

hariombalhara reviewed (changes requested):
Left another review @retrogtx. There are still somet hings pending.

retrogtx committed:
fix: embed type selection

retrogtx committed:
fix: update layout selection in EmbedCodes to use dynamic configuration

retrogtx committed:
fix: update embed configuration to use theme instead of layout

retrogtx committed:
fix: conditionally render embed options based on if we are selecting atom

hariombalhara commented on packages/features/embed/lib/EmbedCodes.tsx:
```diff
@@ -97,21 +97,138 @@ export const Codes = {
+      namespace,
+    }: {
+      calLink: string;
+      uiInstructionCode: string;
```
nit: unused variable

hariombalhara commented on packages/features/embed/lib/EmbedTabs.tsx:
```diff
@@ -123,6 +123,59 @@ export const tabs = [
+        <>
+          <small className="text-subtle flex py-2">{t("create_update_react_component")}</small>
+          <div className="text-default bg-attention mb-3 rounded-md p-3 text-sm">
+            <strong>Note:</strong> The preview shown in the Preview tab is generated using the iframe
```
Nit: Use translation

hariombalhara commented on packages/ui/components/navigation/tabs/HorizontalTabItem.tsx:
```diff
@@ -33,9 +33,25 @@ const HorizontalTabItem = function ({
+  const urlMatches = useUrlMatchesCurrentUrl(href, matchFullPath);
   const { t } = useLocale();
 
+  let isCurrent = props?.isActive;
+  if (isCurrent === undefined) {
+    const isEmbedTab = href.includes("embedTabName=");
+    if (isEmbedTab) {
+      const urlSearchParams = new URLSearchParams(window.location.search);
+      const currentEmbedTabName = urlSearchParams.get("embedTabName");
+
+      const embedTabNameMatch = href.match(/embedTabName=([^&]*)/);
+      const tabEmbedTabName = embedTabNameMatch ? embedTabNameMatch[1] : null;
+
+      isCurrent = currentEmbedTabName === tabEmbedTabName;
+    } else {
+      isCurrent = urlMatches;
+    }
+  }
```
This isn;t right approach as the HorizontalTabItem component is a a generic one and not specific to embeds.

Can we find a way to fix the bug by not having any embed specific logic here.

retrogtx committed:
Merge branch 'main' into embed-gen

retrogtx committed:
fix: use simple props instead of embed specific logic to fix the active embed bug

hariombalhara committed:
Fix double tabs activeness

hariombalhara committed:
Revert isCurrent change

hariombalhara reviewed (dismissed):
LGTM now !! Great work @retrogtx and super quick iterations !!

hariombalhara committed:
Remove isCurrent from types
